root/usr/share/fdi/facts/discovery-facts.rb

```diff
@@ -109,34 +109,52 @@ def discovery_bootif
 has_ipmi = $?.exitstatus == 0
 
 if has_ipmi
-  output = Facter::Util::Resolution.exec("ipmitool lan print 1 2>/dev/null")
-  attributes = {}
-  output.each_line do |line|
-    case line.strip
-    when /^IP Address Source\s+: (.*)/
-      attributes[:ipaddress_source] = $1
-    when /^IP Address\s+: (.*)/
-      attributes[:ipaddress] = $1
-    when /^Subnet Mask\s+: (.*)/
-      attributes[:subnet_mask] = $1
-    when /^MAC Address\s+: (.*)/
-      attributes[:macaddress] = $1
-    when /^Default Gateway IP\s+: (.*)/
-      attributes[:gateway] = $1
+  Facter.add("ipmi_enabled") do
+    confine :kernel => "Linux"
+    setcode do
+      true
     end
-  end if output
-
-  if attributes.keys.empty?
-    Facter.debug("Running ipmitool didn't give any information")
   end
-  attributes[:enabled] = true
-  attributes.each do |fact, value|
-    Facter.add("ipmi_#{fact}") do
-      confine :kernel => "Linux"
-      setcode do
-        value
+  default_found = false
+  (0..15).each do |n|
+    output = Facter::Util::Resolution.exec("ipmitool lan print #{n} 2>/dev/null")
+    attributes = {}
+    output.each_line do |line|
+      case line.strip
+      when /^IP Address Source\s+: (.*)/
+        attributes[:ipaddress_source] = $1
+      when /^IP Address\s+: (.*)/
+        attributes[:ipaddress] = $1
+      when /^Subnet Mask\s+: (.*)/
+        attributes[:subnet_mask] = $1
+      when /^MAC Address\s+: (.*)/
+        attributes[:macaddress] = $1
+      when /^Default Gateway IP\s+: (.*)/
+        attributes[:gateway] = $1
+      end
+    end if output
+
+    if attributes.keys.empty?
+      Facter.debug("Running ipmitool didn't give any information")
```

Comment:
Yeah, actually I've changed it and only report when the first channel was not yet found.